uswid/identity.py

```diff
@@ -155,6 +156,101 @@ def import_xml(self, xml: bytes) -> None:
             link._import_xml(node)
             self.add_link(link)
 
+    def import_json(self, blob: bytes) -> None:
+        """imports a SWID JSON blob"""
+
+        try:
+            identity = json.loads(blob)
+        except json.decoder.JSONDecodeError as e:
+            raise NotSupportedError("invalid JSON: {}".format(e)) from e
+
+        # identity
+        self.tag_id = identity.get("tag-id")
+        self.tag_version = identity.get("tag-version")
+        self.software_name = identity.get("software-name")
+        self.software_version = identity.get("software-version")
+
+        # optional metadata
+        for meta in identity["software-meta"]:
+            for attr_name, attrib_name in [
+                ("summary", "summary"),
+                ("revision", "revision"),
+                ("product", "product"),
+                ("edition", "edition"),
+                ("colloquial-version", "colloquial_version"),
+            ]:
+                if attr_name in meta:
+                    setattr(self, attrib_name, identity["software-meta"][attr_name])
```

Comment:
the CoSWID spec says: 'software-meta => one-or-more<software-meta-entry>' (page 13)
I honestly don't know, in which case one would want more than one software-meta-entry, but let's stay compliant to the spec (it doesn't complicate stuff much anyway).

Reply:
It's the same problem with the 'entities'. Since .ini files don't have an array like structure, one can either bypass the problem like it's currently the case with the entities or only allow converting from .ini to all other formats, but not the other way around.

Reply:
I honestly think making software-meta an array is going to be a future foot-gun. It's also one more thing to explain.

Reply:
I would really really like to stay compliant to the spec, otherwise there will be a hell of incompatibilities between different tools implementing the CoSWID standard. But in order to make the .ini format still usable, I would suggest that .ini files only allow 1 software-meta (which is what you want 99% of all times). But I would still use/allow arrays inside the JSON XML and CBOR format, since that is required by the CoSWID spec. That way we stay compliant with the spec, but can still write .ini files in a normal fashion.

Reply:
Okay, I'm not super passionate about this, so I've added the array.